### DISCUSSION TITLE
lsp框架导致无法开机

### DISCUSSION THREAD

**Discussion by HollisMeynell** [LSPosed/LSPosed#1846] [Q&A] [answered]
进入adb shell删除/data/adb/modules/zygisk_lsposed 尝试删除模块无效仍旧卡米,重启会恢复所有文件
![image](https://user-images.githubusercontent.com/56613764/163356355-ea052ef1-299b-4959-bc5d-e0ccc58a2df3.png)
尝试刷回初始boot分区卸载magisk可以正常开机,但是丢失root权限无法进行任何卸载操作,无法打开lsposed以及magisk无法操作模块
尝试刷机全量包并重新获取root失败,刷入magisk boot.img便无法开机
暂无第三方Recovery操作,刷入magisk卡米界面可以使用adb并有su权限,但是无尽加载,恢复boot则可以开机但是没有权限操作,请问如何拯救

**Comment by JasonKhew96** [comment, marked as answer]
正确做法应该是 创建 `disable` (禁用) 或 `remove` (卸载/移除) 空文件到 `zygisk_lsposed` 文件夹里

详情见:
https://topjohnwu.github.io/Magisk/guides.html

**Reply by HollisMeynell** [reply]
感谢您的回答,不过是否我操作不对,似乎没有生效
![image](https://user-images.githubusercontent.com/56613764/163361023-24ad16b6-5a7d-46aa-83c6-305483190a35.png)
/data/adb/modules/zygisk_lsposed此文件夹下存在disable文件夹
我尝试创建remove文件但是没有变化,依然卡在开机界面
我尝试了一下执行uninstall.sh似乎也没有作用
![image](https://user-images.githubusercontent.com/56613764/163361993-fdaf519a-423a-465f-b1f7-1aaae91b970f.png)
(创建文件以及文件夹均无生效)
以及其他模块似乎也不可以
![image](https://user-images.githubusercontent.com/56613764/163363533-2df6cc88-9f84-43b4-8c82-71f6e43fa753.png)
您发的连接我大概读懂是构建magisk模块的结构文档,本人比较笨并没有找的关于卸载模块的命令操作,或许magisk alpha版本发生了变化,但是我搜索到的信息
`rm -r data/adb/modules/module_name`
也无效

**Reply by GithubID-ohh** [reply]
> 正确做法应该是 创建 `disable` (禁用) 或 `remove` (卸载/移除) 空文件到 `zygisk_lsposed` 文件夹里
> 
> 详情见: https://topjohnwu.github.io/Magisk/guides.html

正解! 成功开机😊

**Comment by HollisMeynell** [comment]
补充 手机型号小米12 miui13.0.28  magisk alpha24306版本
安装LSPosed开始可用,重启后无法开机

**Reply by bofuqiang** [reply]
> 补充 手机型号小米12 miui13.0.28 magisk alpha24306版本 安装LSPosed开始可用,重启后无法开机

碰到了同样的问题。我的是摩托罗拉的手机，

**Reply by bofuqiang** [reply]
我找到了解决办法，用原版的boot.img文件引导开机。正常开机后，长按电源键，在弹出的关机按钮上长按，确认重启进入安全模式，重启后，进入bootloader，用fastboot刷入patch后的image，引导进入安全模式。用ADB 执行  shell magisk --remove-modules 即可
